src/interval/index.ts

```diff
@@ -29,40 +19,48 @@ export function interval<S extends unknown, F extends unknown>({
 
   const $notRunning = $isRunning.map((running) => !running);
 
-  const saveTimeout = createEvent<NodeJS.Timeout>();
+  const saveTimeout =
+    createEvent<{ timeoutId: NodeJS.Timeout; reject: () => void }>();
   const $timeoutId = createStore<NodeJS.Timeout | null>(null).on(
     saveTimeout,
-    (_, id) => id,
+    (_, { timeoutId }) => timeoutId,
   );
-  const saveReject = createEvent<() => void>();
   // eslint-disable-next-line @typescript-eslint/no-empty-function
-  const $rejecter = createStore<() => void>(() => {}).on(saveReject, (_, rj) => rj);
+  const $rejecter = createStore<() => void>(() => {}).on(
+    saveTimeout,
+    (_, { reject }) => reject,
+  );
 
-  const timeoutFx = createEffect<number, void>((timeout) => {
-    return new Promise((resolve, reject) => {
-      const timeoutId = setTimeout(resolve, timeout);
-      saveTimeout(timeoutId);
-      saveReject(reject);
-    });
+  const timeoutFx = attach({
+    source: { timeout: $timeout, running: $isRunning },
+    effect: ({ timeout, running }) => {
+      if (!running) {
+        return Promise.reject();
+      }
+
+      return new Promise((resolve, reject) => {
+        const timeoutId = setTimeout(resolve, timeout);
+        saveTimeout({ timeoutId, reject });
+      });
+    },
   });
 
   const cleanupFx = attach({
-    source: [$timeoutId, $rejecter],
-    effect: ([id, rj]) => {
-      rj();
-      if (id) clearTimeout(id);
+    source: { timeoutId: $timeoutId, rejecter: $rejecter },
+    effect: ({ timeoutId, rejecter }) => {
+      rejecter();
+      if (timeoutId) clearTimeout(timeoutId);
     },
   });
 
-  guard({
+  sample({
     clock: start,
     source: $timeout,
-    filter: $notRunning,
     target: timeoutFx,
   });
 
   if (leading) {
-    guard({
+    sample({
```

Comment:
This change means that minimal required version of `effector` should be up to `22.2.0` 🤔 
Now is `22.1.2`, which does not have sample with filter yet - looks like this is breaking change and will lead to major update

I suggest to do it as a separate PR and leave guards in place in this one

What you think, @sergeysova ?

Reply:
I've put everything back the way it was before